src/bin/spin.rs

```diff
@@ -44,8 +44,7 @@ async fn _main() -> anyhow::Result<()> {
     let telemetry_config = match spin_telemetry::config::Config::from_env(VERSION.to_string()) {
         Ok(config) => config,
         Err(err) => {
-            eprintln!("Failed to load telemetry config: {err}");
-            return Ok(());
+            return Err(anyhow::anyhow!("Failed to load telemetry config: {err}"));
```

Comment:
A shortcut if you want it:

```suggestion
            anyhow::bail!("Failed to load telemetry config: {err}"));
```

(definitely not important)

Reply:
Also, would `.context("Failed to load telemetry config")` be appropriate here?